llarp/ev/ev.cpp

```diff
@@ -143,13 +143,54 @@ llarp_ev_udp_sendto(struct llarp_udp_io *udp, const sockaddr *to,
 }
 
 #ifndef _WIN32
+#include <string.h>
+
 bool
 llarp_ev_add_tun(struct llarp_ev_loop *loop, struct llarp_tun_io *tun)
 {
+  if(strcmp(tun->ifaddr, "") == 0 || strcmp(tun->ifaddr, "auto"))
+  {
+    std::string ifaddr = llarp::findFreePrivateRange();
+    std::string addr;
+    auto pos = ifaddr.find("/");
+    if(pos == std::string::npos)
+    {
+      llarp::LogWarn("Auto ifaddr didn't return a netmask: ", ifaddr);
+      return false;
+    }
+    int num;
+    std::string part = ifaddr.substr(pos + 1);
+#if defined(ANDROID) || defined(RPI)
+    num = atoi(part.c_str());
+#else
+    num = std::stoi(part);
+#endif
+    if(num <= 0)
+    {
+      llarp::LogError("bad ifaddr netmask value: ", ifaddr);
+      return false;
+    }
+    tun->netmask = num;
+    addr         = ifaddr.substr(0, pos);
+    strcpy(tun->ifaddr, addr.c_str());
+    llarp::LogInfo("IfAddr autodetect: ", tun->ifaddr, "/", tun->netmask);
+  }
+  if(strcmp(tun->ifname, "") == 0 || strcmp(tun->ifname, "auto"))
+  {
+    strcpy(tun->ifname, llarp::findFreeLokiTunIfName().c_str());
```

Comment:
```
std::string ifname = llarp::findFreeLokiTunIfName();
strncpy(tun->ifname, ifname.c_str(), std::min(sizeof(tun->ifname), ifname.size()));
```

Reply:
same as above. `const std::string` if you care enough.